campaigns_api.go

```diff
@@ -0,0 +1,146 @@
+package createsend
+
+import (
+	"errors"
+	"fmt"
+	"github.com/xitonix/createsend/campaigns"
+	"github.com/xitonix/createsend/campaigns/orderfield"
+	"github.com/xitonix/createsend/internal"
+	"github.com/xitonix/createsend/order"
+	"net/url"
+	"time"
+)
+
+type campaignAPI struct {
+	client internal.Client
+}
+
+func newCampaignAPI(client internal.Client) *campaignAPI {
+	return &campaignAPI{client: client}
+}
+
+func (c *campaignAPI) Create(clientID string, campaign campaigns.CampaignFromUrl) (string, error) {
+	path := fmt.Sprintf("campaigns/%s.json", url.QueryEscape(clientID))
+	var cId string
+	err := c.client.Post(path, &cId, campaign)
```

Comment:
We should always explicitly return `nil` (or empty value) if there is an error. This applied to all the methods below
```
if err != nil {
	return "", err
}
OR
if err != nil {
	return nil, err
}
```